### PR TITLE
[Impeller] Document blending/color management

### PR DESCRIPTION
**Pull request by bdero**
Fixes https://github.com/flutter/flutter/issues/105563.

I focused on explaining the whole "pipeline blend" vs "advanced blend" situation and didn't put the equations in-line since it would mostly just be repeating the W3 doc, but I can do so if desirable (perhaps in a follow-up?)

### CODE REVIEW

File: impeller/docs/blending.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+| Source color | Any color that is output by a fragment shader. |
+| Destination color | The backdrop color in a blend operation. |
+| Premultiplied color | A color that has its alpha multiplied into it. Used for additive blending operations as well as colors presented to a surface. |
+| Porter-Duff alpha composite | One of several operations that add together a source color and destination color, with both the source and destination colors being multiplied by respective alpha factors. |
+| Pipeline blend | A blend mode that Impeller can always implement by using the raster pipeline blend configuration provided by the underlying graphics backend. Most of these are simple _Porter-Duff alpha composites_. |
+| Advanced blend | A blend mode that Impeller computes using a fragment program. |
+
+## Premultiplied colors
+
+In impeller, all blending _source colors_ are assumed to be _premultiplied_ for
```
Comment (chinmaygarde) [inline impeller/docs/blending.md:30]: Nit: Impeller

File: impeller/docs/blending.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+All blend modes conform to the
+[W3C Compositing and Blending recommendation](https://www.w3.org/TR/compositing-1/).
+
+Blend operations are driven by the `Entity::BlendMode` enum. In the Aiks layer,
+all drawing operations conform to the given `Paint::blend_mode`. In the Entities
+layer, all Entities have an associated blend mode, which can be set via
+`Entity::SetBlendMode(Entity::BlendMode)`.
+
+## Glossary of blending terms
```
Comment (chinmaygarde) [inline impeller/docs/blending.md:18]: In a later patch perhaps, the performance implications of advanced blends could be added.

**Review by chinmaygarde — APPROVED** [review]
This is great! Some nits but LGTM.

File: impeller/docs/blending.md (hunk 1 above)
Comment (bdero) [reply impeller/docs/blending.md]: fixed

File: impeller/docs/blending.md (hunk 2 above)
Comment (bdero) [reply impeller/docs/blending.md:18]: Added some details about why advanced blends are much slower than pipeline blends.

**Pull request merged**